src/main/java/mat/server/LoggedInUserUtil.java

```diff
@@ -2,178 +2,131 @@
-			}
-			return userName;
-		}
-	}
-	
-	/**
-	 * Sets the logged in login id.
-	 * 
-	 * @param loginid
-	 *            the new logged in login id
-	 */
-	public static void setLoggedInLoginId(String loginid) {
-		loggedInLoginId = loginid;
-	}
-	
-	/**
-	 * Gets the logged in login id.
-	 * 
-	 * @return the logged in login id
-	 */
-	public static String getLoggedInLoginId() {
-		if(loggedInLoginId != null) {
-			return loggedInLoginId;
-		}
-		else {
-			UsernamePasswordAuthenticationToken token = getToken();
-			String loginId = null;
-			if(token != null) {
-				loginId = ((MatUserDetails) token.getDetails()).getLoginId();
-			}
-			return loginId;
-		}
-	}
-	
-	/**
-	 * Sets the logged in user email.
-	 * 
-	 * @param e
-	 *            the new logged in user email
-	 */
-	public static void setLoggedInUserEmail(String e) {
-		loggedInUserEmail = e;
-	}
-	
-	/**
-	 * Gets the logged in user email address.
-	 * 
-	 * @return the logged in user email address
-	 */
-	public static String getLoggedInUserEmailAddress() {
-		if(loggedInUserEmail != null) {
-			return loggedInUserEmail;
-		}
-		UsernamePasswordAuthenticationToken token = getToken();
-		String emailAddress = null;
-		if(token != null) {
-			emailAddress = ((MatUserDetails) token.getDetails()).getEmailAddress();
-			
-		}
-		return emailAddress;
-	}
-	
-	/**
-	 * Gets the logged in user role.
-	 * 
-	 * @return the logged in user role
-	 */
-	public static String getLoggedInUserRole() {
-		String role = null;
-		UsernamePasswordAuthenticationToken token = getToken();
-		if(token != null) {
-			Iterator<GrantedAuthority> iter = token.getAuthorities().iterator();
-			if(iter.hasNext()) {
-				role = iter.next().getAuthority();
-			}
-		}
-		return role;
-	}
-
-	public static void setLoggedInUserName(String e) {
-	      loggedInUserName = e;
-	}
-	
-	
-	public static String getLoggedUserName() {
-		if(loggedInUserName != null) {
-			return loggedInUserName;
-		}
-		else {
-			UsernamePasswordAuthenticationToken token = getToken();
-			String firstName = null;
-			String lastName = null;
-			if(token != null) {
-				firstName = ((MatUserDetails) token.getDetails()).getUsername();
-				lastName = ((MatUserDetails) token.getDetails()).getUserLastName();
-			}
-			return firstName + " " + lastName;
-		}
-	}
+
+    private static final Log logger = LogFactory.getLog(LoggedInUserUtil.class);
+
+    private static String loggedInUser;
+
+    private static String loggedInUserEmail;
+
+    private static String loggedInLoginId;
+
+    private static String loggedInUserName;
+
+    /**
+     * Gets the token.
+     *
+     * @return the token
+     */
+    private static UsernamePasswordAuthenticationToken getToken() {
+        //re-factored to support Anonymous user US 439
+        UsernamePasswordAuthenticationToken token = null;
+        if (SecurityContextHolder.getContext() != null) {
+            if (SecurityContextHolder.getContext().getAuthentication() != null) {

```

Comment:
Can't we use single if statement ?

Reply:
I didn't change this piece of code :) It was just re-formatted. Tabs->Spaces

Reply:
and removed some redundant obsolete comments

If you check "Hide whitespace changes" you will see cleaner PR view

Reply:
ok